src/plugins/elasticsearch/lib/create_proxy.js

```diff
@@ -12,7 +12,8 @@ function createProxy(server, method, route, config) {
         mapUri: mapUri(server),
         passThrough: true,
         agent: createAgent(server),
-        xforward: true
+        xforward: true,
+        timeout: server.config().get('elasticsearch.requestTimeout') + 100
```

Comment:
The race condition is a valid concern, but honestly, if the socket is timing out, I think we should just let it time out and leave it at that. Added any amount of time padding there, there's still no guarantee we're going to see an error, it could still just timeout.